ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp

```diff
@@ -188,13 +198,26 @@ MaxPool2D::MultiCore::cached_program_t max_pool_2d_multi_core_sharded_with_halo_
     auto in_tiled_cb = tt::tt_metal::CreateCircularBuffer(program, all_cores, in_tiled_cb_config);
     log_debug(tt::LogOp, "CB {} :: PS = {}, NP = {}", in_tiled_cb_id, in_tiled_cb_pagesize, in_tiled_cb_npages);
 
+
+    // output of reduce == writer to write
+    uint32_t out_cb_id = tt::CB::c_out0;  // output rows in RM
+    // after reduction
+    uint32_t out_cb_pagesize = output.shard_spec().value().shape[1] * out_nbytes / in_nblocks_c;  // there is just one row of channels after each reduction (or 1 block of c if its greater than 8 tiles)
+    uint32_t out_cb_npages = output.shard_spec().value().shape[0] * in_nblocks_c;
+    CircularBufferConfig cb_out_config = CircularBufferConfig(out_cb_npages * out_cb_pagesize, {{out_cb_id, out_df}})
+                                             .set_page_size(out_cb_id, out_cb_pagesize)
+                                             .set_globally_allocated_address(*output.buffer());
+    ;
+    auto cb_out = tt::tt_metal::CreateCircularBuffer(program, all_cores, cb_out_config);
+    log_debug(tt::LogOp, "CB {} :: PS = {}, NP = {}", out_cb_id, out_cb_pagesize, out_cb_npages);
+
     if (is_large_kernel) {
         uint32_t max_pool_partials_cb_id = tt::CB::c_intermed1;  // max_pool partials
-        uint32_t max_pool_partials_cb_pagesize = in_cb_sz;
+        uint32_t max_pool_partials_cb_pagesize = out_cb_pagesize;
```

Comment:
Minor comment, This variable should be min(out_cb_pagesize, TILE_SIZE*8*nbytes).

Reply:
Hey looks like a small typo, I updated it to ` min(out_cb_pagesize, TILE_SIZE * 8 * out_nbytes);`, let me know if this isn't what you meant!